### PR TITLE
Added csv validation and updated dependencies

### PR DESCRIPTION
**Pull request by charlottekostelic**
Updated pytest and removed py from requirements.txt to address security issue (#6). Also installed pandas-stubs to address typing issues. 

Added validate module to review downloaded csv before creating MARC records (#7). This module validates that the csv headers conform to expectations and the timestamps are formatted correctly.

### CODE REVIEW

**Comment by charlottekostelic** [comment]
Edited download.py so any fields that might be interpreted as floats would be imported as strings. As noted in #10, pandas was importing the "Date of Publication" field as a float which caused the 008 to be too long. 
Also noticed that the 005 was formatted incorrectly as _yymmddhhmmss.f_ so I changed it to _yyyymmddhhmmss.f_.

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@
 4. Activate project's virtual environment
     * `source ./.venv/scripts/activate`
 5. Run the following Python command in terminal (the next control # in the example is 20)
-    * `python bagel/run.py 20`
+    * `python run.py 20`
```
Comment (klinga) [inline README.md:29]: 👍

File: bagel/download.py [hunk 2]
```diff
@@ -11,6 +9,21 @@
 def get_metadata():
     # download metadata from google sheet and save it as a csv
     df = pd.read_csv(
-        URL, usecols=range(0, 25), converters={22: lambda x: x.replace("\n", " ")}
+        URL,
+        usecols=range(0, 25),
+        converters={
```
Comment (klinga) [inline bagel/download.py:14]: Interesting, I initially thought data types should be specified in `dtypes` argument. Looks like `converters` is very flexible way to configure parsing.

File: bagel/produce.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def game_record(data, control_number, suppressed=True, status_code="-"):
 
     # 005
     tags.append(
-        Field(tag="005", data=datetime.strftime(datetime.now(), "%y%m%d%H%M%S.%f"))
+        Field(tag="005", data=datetime.strftime(datetime.now(), "%Y%m%d%H%M%S.%f"))
```
Comment (klinga) [inline bagel/produce.py:76]: Thanks for catching this bug! I wonder if there was a test for it. Probably not....

File: bagel/validate.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        "Date of publication",
+        "Description/summary",
+        "List of components",
+        "Email Address",
+    ]
+    with open("temp/metadata.csv", "r", encoding="utf-8") as csvfile:
+        reader = csv.reader(csvfile)
+        csv_header = next(reader)
+        try:
+            assert csv_header == correct_headings
```
Comment (klinga) [inline bagel/validate.py:38]: Nice use of `assert` in code 👍 
Don't be afraid to use assert in this fashion! `assert` does not only belongs in tests.

File: bagel/validate.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+def validate_timestamps():
+    # test validity of csv by tring to convert str in first column to a timestamp
+    with open("temp/metadata.csv", "r", encoding="utf-8") as csvfile:
+        reader = csv.reader(csvfile)
+        next(reader)
+        for n, row in enumerate(reader):
+            try:
+                datetime.strptime(row[0], "%m/%d/%Y %H:%M:%S")
+            except Exception:
```
Comment (klinga) [inline bagel/validate.py:52]: great!

File: temp/.gitignore
Comment (klinga) [inline temp/.gitignore:1]: I know why you created the `temp` folder - a cleaner solution would be the `tmpdir` fixture in pytest: https://docs.pytest.org/en/6.2.x/tmpdir.html
We can discuss

File: tests/test_validate.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        reader = csv.reader(csvfile)
+        csv_header = next(reader)
+        assert csv_header == correct_headings
+
+
+def test_validate_timestamps():
+    with open("temp/metadata.csv", "r", encoding="utf-8") as csvfile:
+        reader = csv.reader(csvfile)
+        next(reader)
+        for row in reader:
```
Comment (klinga) [inline tests/test_validate.py:43]: A general note, I've done it many times myself, but when you loop over an empty iterable, your `assert` will not hit anything and you may silently pass over some problems.

**Review by klinga — APPROVED** [review]
This was really good! Impressive!

**Pull request merged**